### PR TITLE
Fix CI with backports from master

### PR DESCRIPTION
**Pull request by ShGKme**
Fixes failing CI

Manual backport of:
- https://github.com/nextcloud/spreed/pull/9629
  - Fixes failing CI because of wrong PHP extensions in `Set up php`
- https://github.com/nextcloud/spreed/pull/9666
  - Fixes failing CI because of out of space (there is an alternative solution - use Ubuntu 22.04)
- https://github.com/nextcloud/spreed/pull/9672
  - I'm not sure this is actually required...

### CODE REVIEW

**Comment by ShGKme** [comment]
@skjnldsv We had many problems on CI on stable27:
- With PHP extensions (no GC)
- With out of space

All it works fine (almost) on master. So I decided that it is better to have the same CI on stable27 than its own fixes like using Ubuntu 22.04 or manually change ONLY php extensions.

But I might be wrong if some CI updates in master are actually hard-connected with master and should not be used on 27...

Could you confirm, that this is ok?

File: .github/workflows/appstore-build-publish.yml [hunk 1]
```diff
@@ -83,7 +83,7 @@ jobs:
         if: steps.check_composer.outputs.files_exists == 'true'
         run: |
           cd ${{ env.APP_NAME }}
-          composer install --no-dev --classmap-authoritative
+          composer install --no-dev
```
Comment (SystemKeeper) [inline .github/workflows/appstore-build-publish.yml:86]: We should create a follow up issue to this and clarify (also for master).

**Review by SystemKeeper — APPROVED** [review]
Approving this now as release needs to be done. Can't find anything obviously wrong.

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**